src/vm/amd64/unixasmhelpers.S

```diff
@@ -304,7 +232,7 @@ NESTED_ENTRY OnHijackInteriorPointerTripThread, _TEXT, NoHandler
 
     END_PROLOGUE
 
-    call                C_FUNC(OnHijackInteriorPointerWorker)
+    call                C_FUNC(OnHijackWorker)
 
     free_stack          0x8
     pop_register        rax
```

Comment:
`RDX` and `RAX` are pushed on to the stack by this `OnHijackGCTripThread()` helper.
So, `OnHijackWorker()` and eventually `HijackFrame::GCScanRoots()` will recieve them as `m_Args->ReturnValue[2]`. 
The second register `RDX` will be updated by this case:
https://github.com/dotnet/coreclr/pull/6522/files#diff-bffe9cf484a03aa5d9d19d6842a144b1R1175